File: discorddeputy-api/src/main/java/com/freberg/discorddeputy/DiscordNotification.java
package com.freberg.discorddeputy;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import org.jspecify.annotations.NonNull;

import java.time.Instant;

@JsonIgnoreProperties(ignoreUnknown = true)
public record DiscordNotification(@NonNull String id,
                                  @NonNull String source,
                                  @NonNull String type,
                                  @NonNull Instant timestamp,
                                  @NonNull String title,
                                  String descriptionHeader,
                                  @NonNull String description,
                                  String url,
                                  String imageUrl,
                                  String thumbnailUrl) {
}

File: discorddeputy-fetcher-steam/src/main/kotlin/com/freberg/discorddeputy/DiscordNotification.kt
package com.freberg.discorddeputy

import java.time.Instant

@Suppress("unused")
data class DiscordNotification(
    val id: String,
    val timestamp: Instant,
    val title: String,
    val description: String,
    val url: String,
    val source: String = "steam",
    val type: String = "news"
)